packages/terra-spacer/src/_spacerShorthandUtils.js

```diff
@@ -0,0 +1,99 @@
+import { SpacerSizes } from './_spacerConstants';
+
+const SHORTHAND_DELIMITER = ' ';
+const shLengthBoundaries = { min: 1, max: 4 };
+
+const conflictingPropNamesError = (
+  propName = '<SUPPLY SHORTHAND PROP NAME>',
+  conflictingPropName = '<SUPPLY CONFLICTING PROP NAME>',
+  componentName = 'Spacer',
+) => new Error(`Expected only ${propName} or ${conflictingPropName} to be supplied to ${componentName} but both were provided. Validation failed.`);
```

Comment:
Is there a way to throw a warning here instead of hard crashing the entire page?

Reply:
@bjankord and I actually discussed this [here](https://github.com/cerner/terra-core/issues/1314#issuecomment-377193957).

The prop-types [doc](https://github.com/facebook/prop-types/blame/master/README.md#L118-L128) (sorry for the `blame` view, it's the only way i could find to link line numbers), specifically instructs me not to throw warnings

Reply:
Does returning the error crash the entire page or does returning the error flow through the prop validation and log a warning in the console as a non-custom prop would?

Reply:
It just returns a logged warning.
---
When I do 
![image](https://user-images.githubusercontent.com/17390173/39473289-b7675dac-4d13-11e8-80d1-af8f6ec6e8c5.png)

I get 
![image](https://user-images.githubusercontent.com/17390173/39473297-c40d076e-4d13-11e8-87bb-ccbd5f94bf83.png)

I've purposefully chosen to only throw the warning on the first conflicting prop existence (first defined in the order that CSS maps individual shorthands to directions: top, right, bottom, left). I've done this to allow for short-circuiting the check for other existing conflicting prop existances and just return the error immedietly. If we want to return **all** conflicting props, I can do that, but didn't see much value in it as if we have 1 conflicting existance, we should be telling the user why and seeing the error should be intuitive enough to know that

> hey, i can't mix shorthand and directional properties

Reply:
That makes sense. My concern was throwing an error and crashing the page, but returning the error and seeing the standard warning in the console without a hard crash seems like the appropriate workflow that follows non-custom validations.